tesseract/tesseract_planning/tesseract_process_managers/include/tesseract_process_managers/process_generators/fix_state_collision_process_generator.h

```diff
@@ -100,5 +117,47 @@ class FixStateCollisionProcessGenerator : public ProcessGenerator
 
   void process(ProcessInput input) const;
 };
+
+/**
+ * @brief Checks if a joint state is in collision
+ * @param start_pos Vector that represents a joint state
+ * @param input Process Input associated with waypoint. Needed for kinematics, etc.
+ * @return True if in collision
+ */
+bool StateInCollision(const Eigen::Ref<Eigen::VectorXd>& start_pos,
```

Comment:
I think this can just be `Eigen::Ref<Eigen::VectorXd> start_pos`,

Reply:
Oh. This doesn't have to be writable. Based on the trajopt PR. I think it should be `const Eigen::Ref<const Eigen::VectorXd>& start_pos`

Reply:
Ah, change looks good.